src/Core/MailTemplate/Transformation/MailVariablesTransformation.php

```diff
@@ -26,22 +26,48 @@
 
 namespace PrestaShop\PrestaShop\Core\MailTemplate\Transformation;
 
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+
 /**
  * Class MailVariablesTransformation is used only for preview, it replaces the
  * variables present in the mail templates (this replacement is usually performed
  * by the Mail class in real behavior).
  * You can set the variables using the actionBuildMailLayoutVariables and setting
- * them in the `templateVars` key.
+ * them in the `templateVars` key, or simply via the constructor.
  */
 class MailVariablesTransformation extends AbstractTransformation
 {
+    /**
+     * @var array
+     */
+    private $replaceVariables;
+
+    /**
+     * @param string $type
+     * @param array $replaceVariables
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct($type, array $replaceVariables = array())
```

Comment:
```suggestion
    public function __construct(string $type, array $replaceVariables = [])
```

Reply:
does it work on every PHP versions? or is it fine since we dropped 5.6?

Reply:
oh ok it's since `PHP 7.0.0` so we're good to go!

Reply:
oh but since it already existed in 176 is it a BC?

Reply:
When we'll drop the 5.6 support, it'll not be a BC :)